src/extensions/lens-extension.ts

```diff
@@ -114,7 +114,7 @@ export class LensExtension {
 
   @action
   activate() {
-    return this.onActivate();
+    return Promise.resolve(this.onActivate());
```

Comment:
`onActive` might not return a Promise. This converts non-promises to promises.

Reply:
Does it? If it throws I don't think a promise will be returned

Reply:
Good point, changed this to handle sync errors